src/client/Microsoft.Identity.Client/AppConfig/PublicClientApplicationBuilder.cs

```diff
@@ -277,6 +277,23 @@ public PublicClientApplicationBuilder WithParentActivityOrWindow(Func<IntPtr> wi
         }
 #endif
 
+        /// <summary>
+        /// Returns true if a broker can be used. 
+        /// </summary>
+        /// <remarks>
+        /// On Windows, the broker (WAM) can be used on Win10 and is always installed. See https://aka.ms/msal-net-wam
+        /// On Mac, Linux and older versions of Windows a broker is not available.
+        /// If your application is .NET5, please use the target .net5.0-windows10.0.17763.0 for all Windows versions and target net5.0 to target Linux and Mac.
+        /// If your application is .NET classic or .NET Core 3.1 and you wish to use the Windows Broker, please install Microsoft.Identity.Client.Desktop first and call WithDesktopFeatures().
+        /// 
+        /// On mobile, the device must be Intune joined and Authenticator or Company Portal must be installed. See https://aka.ms/msal-brokers
+        /// </remarks>
+        public bool IsBrokerAvailable()
```

Comment:
@jmprieur @jennyf19 @pmaytak - I moved `IsBrokerAvaiable()` to the builder, instead of on the app. The reason is that `WithBroker` is an app property, so you'd have to create the app via `app = builder.WithBroker()` and then call `app.IsBrokerAvailable()`

The more natural way is to have it on the app builder: 

```csharp
var builder = PublicClientApplicationBuilder().Create("client_id");
if (builder.IsBrokerAvailable()) 
{
 Console.WriteLine("No broker! Install one!"); 
} 
else 
{ 
     builder = builder.WithBroker();
}
}

```

Thx @mjcheetham for pointing out this problem!

Reply:
makes sense. thanks for the heads-up